assets/css/main.scss

```diff
@@ -1,8 +1,139 @@
+    $top-nav-dropdown-link: {{ site.data.theme.colors.top-navigation.top-nav-dropdown-link | default: "$color-white" }};
+    $top-nav-dropdown-link-hover-bg: {{ site.data.theme.colors.top-navigation.top-nav-dropdown-link-hover-bg | default: "$color-transparent" }};
+  {% endif %}
+
+  // Side nav
+  {% if site.data.theme.colors.side-navigation %}
+    $side-nav-bg: {{ site.data.theme.colors.side-navigation.side-nav-bg | default: "$color-transparent" }};
+    $side-nav-link: {{ site.data.theme.colors.side-navigation.side-nav-link | default: "$color-base" }};
+    $side-nav-link-hover: {{ site.data.theme.colors.side-navigation.side-nav-link-hover | default: "$color-blue" }};
+    $side-nav-link-hover-bg: {{ site.data.theme.colors.side-navigation.side-nav-link-hover-bg | default: "$color-gray-lightest" }};
+    $side-nav-link-current: {{ site.data.theme.colors.side-navigation.side-nav-link-current | default: "$color-blue" }};
+  {% endif %}
+
+  // Color - Hero
+  {% if site.data.theme.colors.hero %}
+    $hero-bg: {{ site.data.theme.colors.hero.hero-bg | default: "$color-primary-darkest" }};
+    $hero-header-alt: {{ site.data.theme.colors.hero.hero-header-alt | default: "$color-white" }};
+    $hero-header: {{ site.data.theme.colors.hero.hero-header | default: "$color-aqua" }};
+    $hero-text: {{ site.data.theme.colors.hero.hero-text | default: "$color-white" }};
+    $hero-link: {{ site.data.theme.colors.hero.hero-link | default: "$color-aqua" }};
+    $hero-button-bg: {{ site.data.theme.colors.hero.hero-button-bg | default: "$color-blue" }};
+    $hero-button-text: {{ site.data.theme.colors.hero.hero-button-text | default: "$color-white" }};
+  {% endif %}
+
+  // Color - Alt section
+  {% if site.data.theme.colors.alt-section %}
+    $alt-section-bg: {{ site.data.theme.colors.alt-section.bg-color | default: "$color-primary-darkest" }};
+    $alt-section-header: {{ site.data.theme.colors.alt-section.header-color | default: "$color-aqua" }};
+    $alt-section-text: {{ site.data.theme.colors.alt-section.text-color | default: "$color-white" }};
+    $alt-section-link: {{ site.data.theme.colors.alt-section.link-color | default: "$color-gray-lighter" }};
+    $alt-section-link-hover: {{ site.data.theme.colors.alt-section.link-hover-color | default: "$color-white" }};
+  {% endif %}
+
+  // Color - Footer
+  {% if site.data.theme.colors.footer %}
+    $primary-footer-bg: {{ site.data.theme.colors.footer.primary-footer-bg | default: "$color-gray-lightest" }};
+    $primary-footer-link: {{ site.data.theme.colors.footer.primary-footer-link | default: "$color-base" }};
+    $footer-bg: {{ site.data.theme.colors.footer.footer-bg | default: "$color-gray-lighter" }};
+    $footer-text: {{ site.data.theme.colors.footer.footer-text | default: "$color-base" }};
+    $footer-link: {{ site.data.theme.colors.footer.footer-link | default: "$color-base" }};
+  {% endif %}
+
+{% endif %}
+
 @import 'uswds/all';
+
+{% if site.data.theme.colors.usa-banner %}
```

Comment:
That is a great question! Each variable is defaulted to use the color from the uswds. So if nothing is defined it will use uswds otherwise it will use whatever colors are defined.

For example, take a look at the bg color here.

```
$usa-banner-bg: {{ site.data.theme.colors.usa-banner.usa-banner-bg | default: "$color-gray-lightest" }};```

Reply:
Yep, totally get that we have fallback variables for the colors, my question in why are the component css files being dynamically imported only when those variables are defined in the config file?

For example, if there are always definitions for the `usa-banner` variables, why is there a conditional loading statement

```
{% if site.data.theme.colors.usa-banner %}
   @import 'components/usa-banner';
{% endif %}
```

Reply:
By default the theme.yml is commented out so there is not always definitions for `usa-banner` variables.
  
The variables for `usa-banner` found in _data/theme.yml are only needed if that section in the yml file is uncommented or active.

Reply:
So users won't be able to get access to any of these components without defining at least one color from each? What if they are just totally fine with the defaults?

Reply:
Exactly! If they are fine with the uswds defaults then they don't have to do anything

Reply:
I'm sorry to beat a dead horse here, but I'm not understanding something in this exhange!

You said `Exactly! If they are fine with the uswds defaults then they don't have to do anything`, but my question was:

`If the user does NOT define any theme, will they lose access to the build in components?`

Based on what I see in this file, it seems like that is the case?

Assuming we're talking about the `usa-banner` component:

* user does *NOT* define any themes/colors/whatever
* the `if` statement, in this file, that controls loading the `usa-banner` CSS, sees that nothing is defined
* the component file is therefore not loaded
* with no CSS, the user can't properly use the component with the defaults

Reply:
Sorry about the confusion, 😖 I think this has been resolved by @hursey013 since the original PR. 

If I remove the `theme.yml` file completely the `usa-banner.scss` file will still be loaded but would be empty due to `@if variable-exists` being used within that file.